learn/architecture.md

```diff
@@ -53,5 +53,67 @@ When the contract is reposted, some of the faulty node's collateral pays for the
 
 ![architect](/learn/architecture.png)
 
+### Marketplace architecture ###
+
+The marketplace consists of a smart contract that is deployed on-chain, and the sales and purchasing modules that are part of the node software. The sales module is responsible for posting storage requests to the smart contract. The purchasing module is its counterpart that storage providers use to determine which storage requests they are interested in.
+
+#### Smart contract ####
+
+The smart contract facilitates matching between storage providers and storage
+clients. A storage client can request a certain amount of storage for a certain
+duration. This request is then posted on-chain, so that storage providers can
+see it, and decide whether they want to fulfill a slot in the request.
+
+The main parameters of a storage request are:
+- the amount of bytes of storage that is requested
+- a content identifier (CID) of the data that should be stored
+- the duration for which the data should be stored
+- the number of slots (based on the erasure coding parameters)
+- an amount of tokens to pay for the storage
```

Comment:
I'm assuming this leaves out all the parameters on purpose?

Reply:
Yes, I wanted to capture the essence not the details.